infra/cache/kInfraRedisCacheWrapper.php

```diff
@@ -0,0 +1,313 @@
+<?php
+
+require_once(dirname(__FILE__) . '/kInfraBaseCacheWrapper.php');
+
+/**
+ * @package infra
+ * @subpackage cache
+ */
+class kInfraRedisCacheWrapper extends kInfraBaseCacheWrapper
+{
+	protected $hostName;
+	protected $port;
+	protected $scheme;
+	protected $timeout;
+	protected $statsKey;
+
+	protected $redis = null;
+	protected $gotError = false;
+	protected static $_stats = array();
+	protected $connectAttempts = 0;
+
+	const STAT_CONN = 'conn';
+	const STAT_OP = 'op';
+	const STAT_COUNT = 'count';
+	const STAT_TIME = 'time';
+	const MAX_CONNECT_ATTEMPTS = 4;
+
+	/* (non-PHPdoc)
+     * @see kBaseCacheWrapper::doInit()
+     */
+	protected function doInit ($config)
+	{
+		if (!isset($config['host']) || !isset($config['port']))
+		{
+			self::safeLog("Missing host or port in config, can't connect without it");
+			return false;
+		}
+
+		require './../../vendor/predis/autoload.php';
+
+		$this->scheme = $config['scheme'];
+		$this->hostName = $config['host'];
+		$this->port = $config['port'];
+		$this->timeout = $config['timeout'];
+
+		$redis = new Redis();
+		try
+		{
+			$redis->connect($this->hostName,$this->port,$this->timeout);
+		}
+		catch(Exception $e)
+		{
+			self::safeLog("failed to connect to redis");
+		}
+
+		$connection = $redis->isConnected();
+
+		if (!$connection)
+		{
+			self::safeLog("failed to connect to redis");
+			return false;
+		}
+
+		$this-> redis = $redis;
+		return true;
+	}
+
+	public function __destruct()
+	{
+		$this->close();
+	}
+
+	protected function close()
+	{
+		if ($this->redis)
+		{
+			$this->redis->close();
+		}
+		$this->redis = null;
+	}
+
+	/* (non-PHPdoc)
+     * @see kBaseCacheWrapper::get()
+     */
+	protected function doGet ($key)
+	{
+		return $this->callAndDetectErrors(function($key) {
+			return $this->redis->get($key);
+		}, array($key));
+	}
+
+	/* (non-PHPdoc)
+     * @see kBaseCacheWrapper::set()
+     */
+	protected function doSet ($key, $var, $expiry = 0)
+	{
+		return $this->callAndDetectErrors(function($key, $var, $expiry = 0) {
```

Comment:
NEed to handle the case where expiry is 0 which is different than how Memcache implements it.